### PR TITLE
feat: FluentVersion object in get_fluent_version()

### PR DESCRIPTION
**Pull request by hpohekar**
Return `FluentVersion` object in `get_fluent_version()`.

```
>>> from ansys.fluent.core import launch_fluent  
>>> solver = launch_fluent()
>>> solver.get_fluent_version()
<FluentVersion.v242: '24.2.0'>
>>> solver.get_fluent_version().value
'24.2.0'
```

```
>>> meshing = launch_fluent(mode="meshing")
>>> meshing.get_fluent_version()
<FluentVersion.v242: '24.2.0'>
>>> meshing.get_fluent_version().value
'24.2.0'
```

### CODE REVIEW

File: src/ansys/fluent/core/session.py [hunk 1]
```diff
@@ -260,7 +261,7 @@ def execute_tui(self, command: str) -> None:
 
     def get_fluent_version(self):
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/session.py:262]: please add a type hint for the return type

File: src/ansys/fluent/core/systemcoupling.py [hunk 2]
```diff
@@ -44,7 +44,7 @@ class SystemCoupling:
     def __init__(self, solver):
         self._solver = solver
         # version check - this requires Fluent 2024 R1 or newer.
-        current_fluent_version = FluentVersion(self._solver.get_fluent_version())
+        current_fluent_version = FluentVersion(self._solver.get_fluent_version().value)
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/systemcoupling.py:47]: `get_fluent_version()` is already returning a `FluentVersion`.

```suggestion
        current_fluent_version = self._solver.get_fluent_version()
```

File: src/ansys/fluent/core/session_pure_meshing.py [hunk 3]
```diff
@@ -55,7 +55,7 @@ def __init__(
         self._base_meshing = BaseMeshing(
             self.execute_tui,
             fluent_connection,
-            self.get_fluent_version(),
+            self.get_fluent_version().value,
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/session_pure_meshing.py:58]: The point is to just use the object rather than the string.

File: tests/test_cad_to_post_ftm.py [hunk 4]
```diff
@@ -433,7 +433,7 @@ def test_exhaust_system(new_fault_tolerant_workflow_session, exhaust_system_geom
     ###############################################################################
     # Check the mesh in Meshing mode
     # TODO: Remove the if condition after a stable version of 23.1 is available and update the commands as required.
-    if float(meshing_session.get_fluent_version()[:-2]) < 23.0:
+    if float(meshing_session.get_fluent_version().value[:-2]) < 23.0:
```
Comment (seanpearsonuk) [inline tests/test_cad_to_post_ftm.py:436]: Use object-based comparison.

File: tests/integration/test_optislang/test_optislang_integration.py [hunk 5]
```diff
@@ -133,7 +133,7 @@ def test_generate_read_mesh(mixing_elbow_geometry):
         os.mkdir(temporary_resource_path)
 
     # TODO: Remove the if condition after a stable version of 23.1 is available and update the commands as required.
-    if float(meshing.get_fluent_version()[:-2]) < 23.0:
+    if float(meshing.get_fluent_version().value[:-2]) < 23.0:
```
Comment (seanpearsonuk) [inline tests/integration/test_optislang/test_optislang_integration.py:136]: Use object-based comparison.

File: tests/integration/test_optislang/test_optislang_integration.py [hunk 6]
```diff
@@ -49,7 +49,7 @@ def test_simple_solve(load_mixing_elbow_param_case_dat):
     solver_session.tui.file.read_case(case_path)
 
     # TODO: Remove the if condition after a stable version of 23.1 is available and update the commands as required.
-    if float(solver_session.get_fluent_version()[:-2]) < 23.0:
+    if float(solver_session.get_fluent_version().value[:-2]) < 23.0:
```
Comment (seanpearsonuk) [inline tests/integration/test_optislang/test_optislang_integration.py:52]: Use object-based comparison.

File: src/ansys/fluent/core/utils/fluent_version.py [hunk 7]
```diff
@@ -30,7 +30,7 @@ def get_version(session=None):
             return image_tag.lstrip("v")
         session = pyfluent.launch_fluent(mode="solver")
 
-    return session.get_fluent_version()
+    return session.get_fluent_version().value
 
 
 def get_version_for_file_name(version: Optional[str] = None, session=None):
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/utils/fluent_version.py:36]: I believe that this function should also return an object.

File: src/ansys/fluent/core/utils/search.py [hunk 8]
```diff
@@ -60,11 +60,11 @@ def _get_version_path_prefix_from_obj(obj: Any):
     prefix = None
     if isinstance(obj, PureMeshing):
         path = ["<meshing_session>"]
-        version = get_version_for_file_name(obj.get_fluent_version())
+        version = get_version_for_file_name(obj.get_fluent_version().value)
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/utils/search.py:63]: these calls can pass in and return objects

File: tests/test_cad_to_post_ftm.py [hunk 9]
```diff
@@ -456,7 +456,7 @@ def test_exhaust_system(new_fault_tolerant_workflow_session, exhaust_system_geom
     # Set the velocity and turbulence boundary conditions for the first inlet
     # (inlet-1).
     # TODO: Remove the if condition after a stable version of 23.1 is available and update the commands as required.
-    if float(meshing_session.get_fluent_version()[:-2]) < 23.0:
+    if float(meshing_session.get_fluent_version().value[:-2]) < 23.0:
```
Comment (seanpearsonuk) [inline tests/test_cad_to_post_ftm.py:459]: Use object-based comparison.

File: tests/test_cad_to_post_wtm.py [hunk 10]
```diff
@@ -129,7 +129,7 @@ def test_mixing_elbow(new_watertight_workflow_session, mixing_elbow_geometry):
     ###############################################################################
     # Check the mesh in Meshing mode
     # TODO: Remove the if condition after a stable version of 23.1 is available and update the commands as required.
-    if float(meshing_session.get_fluent_version()[:-2]) < 23.0:
+    if float(meshing_session.get_fluent_version().value[:-2]) < 23.0:
```
Comment (seanpearsonuk) [inline tests/test_cad_to_post_wtm.py:132]: Use object-based comparison.

File: tests/test_cad_to_post_wtm.py [hunk 11]
```diff
@@ -157,7 +157,7 @@ def test_mixing_elbow(new_watertight_workflow_session, mixing_elbow_geometry):
     # Set up the cell zone conditions for the fluid zone (elbow-fluid). Select
     # water-liquid from the Material list.
     # TODO: Remove the if condition after a stable version of 23.1 is available and update the commands as required.
-    if float(solver_session.get_fluent_version()[:-2]) < 23.0:
+    if float(solver_session.get_fluent_version().value[:-2]) < 23.0:
```
Comment (seanpearsonuk) [inline tests/test_cad_to_post_wtm.py:160]: Use object-based comparison.

File: tests/test_flobject.py [hunk 12]
```diff
@@ -667,7 +667,7 @@ def test_accessor_methods_on_settings_object(load_static_mixer_settings_only):
     modified = velocity_inlet.user_creatable()
     assert existing == modified
 
-    if solver.get_fluent_version() < "24.2.0":
+    if solver.get_fluent_version().value < "24.2.0":
```
Comment (seanpearsonuk) [inline tests/test_flobject.py:670]: Use object-based comparison.

File: tests/test_flobject.py [hunk 13]
```diff
@@ -708,7 +708,7 @@ def test_accessor_methods_on_settings_object(load_static_mixer_settings_only):
     solver.results.graphics.mesh["mesh-1"].rename("mesh_new")
     assert solver.results.graphics.mesh.get_object_names() == ["mesh_new"]
 
-    if solver.get_fluent_version() >= "24.2.0":
+    if solver.get_fluent_version().value >= "24.2.0":
```
Comment (seanpearsonuk) [inline tests/test_flobject.py:711]: Use object-based comparison.

File: tests/test_flobject.py [hunk 14]
```diff
@@ -725,7 +725,7 @@ def test_accessor_methods_on_settings_object_types(load_static_mixer_settings_on
     accuracy_control = (
         solver.setup.models.discrete_phase.numerics.tracking.accuracy_control
     )
-    if solver.get_fluent_version() < "24.1.0":
+    if solver.get_fluent_version().value < "24.1.0":
```
Comment (seanpearsonuk) [inline tests/test_flobject.py:728]: Use object-based comparison.

File: tests/test_flobject.py [hunk 15]
```diff
@@ -799,7 +799,7 @@ def test_find_children_from_fluent_solver_session(load_static_mixer_settings_onl
         if path.endswith("geom_dir_spec")
     )
 
-    if load_static_mixer_settings_only.get_fluent_version() < "24.2.0":
+    if load_static_mixer_settings_only.get_fluent_version().value < "24.2.0":
```
Comment (seanpearsonuk) [inline tests/test_flobject.py:802]: Use object-based comparison.

File: tests/test_fluent_session.py [hunk 16]
```diff
@@ -21,7 +21,7 @@
 
 def _read_case(session, lightweight_setup=True):
     case_path = download_file("Static_Mixer_main.cas.h5", "pyfluent/static_mixer")
-    fluent_version_str = session.get_fluent_version()[:-2]
+    fluent_version_str = session.get_fluent_version().value[:-2]
```
Comment (seanpearsonuk) [inline tests/test_fluent_session.py:24]: Why are we slicing the string? Why don't we do object-based comparison below?

File: tests/test_launcher.py [hunk 17]
```diff
@@ -140,7 +140,7 @@ def test_case_load():
     # Case loaded
     assert session.setup.boundary_conditions.is_active()
     # Mesh available because not lightweight
-    fluent_version = float(session.get_fluent_version()[:-2])
+    fluent_version = float(session.get_fluent_version().value[:-2])
```
Comment (seanpearsonuk) [inline tests/test_launcher.py:143]: This is also unnecessary.

File: tests/test_launcher.py [hunk 18]
```diff
@@ -183,7 +183,7 @@ def test_case_data_load():
...
```
Comment (seanpearsonuk) [inline tests/test_launcher.py:186]: This is also unnecessary.

File: tests/test_meshing_workflow.py [hunk 19]
```diff
@@ -120,7 +120,7 @@ def test_mixing_elbow_meshing_workflow(
     ###############################################################################
     # Check the mesh in Meshing mode
     # TODO: Remove the if condition after a stable version of 23.1 is available and update the commands as required.
-    if float(meshing_session.get_fluent_version()[:-2]) < 23.0:
+    if float(meshing_session.get_fluent_version().value[:-2]) < 23.0:
```
Comment (seanpearsonuk) [inline tests/test_meshing_workflow.py:123]: Should be able to do an object-based comparison.

File: tests/test_solvermode/test_calculationactivities.py [hunk 20]
```diff
@@ -7,7 +7,7 @@ def test_solver_calculation(load_mixing_elbow_mesh):
     scheme_eval = solver_session.scheme_eval.scheme_eval
     assert scheme_eval("(client-get-var 'residuals/plot?)") == True
     # TODO: Remove the if condition after a stable version of 23.1 is available and update the commands as required.
-    if float(solver_session.get_fluent_version()[:-2]) < 23.0:
+    if float(solver_session.get_fluent_version().value[:-2]) < 23.0:
```
Comment (seanpearsonuk) [inline tests/test_solvermode/test_calculationactivities.py:10]: objects

File: tests/test_streaming_services.py [hunk 21]
```diff
@@ -47,7 +47,7 @@ def test_transcript(new_solver_session):
         total_checked_transcripts += int(transcript_checked)
         total_passed_transcripts += int(transcript_passed)
 
-    if solver.get_fluent_version() >= "23.2.0":
+    if solver.get_fluent_version().value >= "23.2.0":
```
Comment (seanpearsonuk) [inline tests/test_streaming_services.py:50]: objects

**Comment by seanpearsonuk** [comment]
Added @iboyd-ansys and @ochernuk so that the system coupling integration can be checked when the PR is ready.

File: src/ansys/fluent/core/systemcoupling.py [hunk 22]
```diff
@@ -44,10 +44,9 @@ class SystemCoupling:
     def __init__(self, solver):
         self._solver = solver
         # version check - this requires Fluent 2024 R1 or newer.
-        current_fluent_version = FluentVersion(self._solver.get_fluent_version())
-        if current_fluent_version < FluentVersion.v241:
+        if self._solver.get_fluent_version() < FluentVersion.v241:
             raise RuntimeError(
-                f"Fluent version is {current_fluent_version.value}. PySystemCoupling integration requires Fluent 24.1.0 or later."
+                f"Fluent version is {self._solver.get_fluent_version().value}. PySystemCoupling integration requires Fluent 24.1.0 or later."
```
Comment (ochernuk) [inline src/ansys/fluent/core/systemcoupling.py:49]:
```suggestion
                f"Fluent version is {self._solver.get_fluent_version().value}. PySystemCoupling integration requires Fluent {FluentVersion.v241.value} or later."
```

**Review by ochernuk — APPROVED** [review]

**Comment by mkundu1** [comment]
We can get the following behaviour by adding a `__repr__` method to `FluentVersion`:
```
>>> solver.get_fluent_version()
"24.2.0"
```

File: src/ansys/fluent/core/utils/fluent_version.py (hunk 7 above)
Comment (mkundu1) [inline src/ansys/fluent/core/utils/fluent_version.py:36]: I'll look into if this and the previous function can some property of `FluentVersion` class (in a separate PR).

**Review by mkundu1 — APPROVED** [review]

**Review by ochernuk — APPROVED** [review]

**Review by prmukherj — APPROVED** [review]

**Comment by hpohekar** [comment]
> We can get the following behaviour by adding an `__repr__` method to `FluentVersion`:
> 
> ```
> >>> solver.get_fluent_version()
> '24.2.0'
> ```

Done. Thank you.
